src/Diagnostics/FxCop/Core/Design/CA1052DiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,203 @@
+    /// </para>
+    /// <para>
+    /// This analyzer behaves as similarly as possible to the existing implementations of the FxCop
+    /// rules, even when those implementations appear to conflict with the MSDN documentation of
+    /// those rules. For example, like FxCop, this analyzer emits a diagnostic when it detects a
+    /// static holder class that is declared "sealed", even though the documentation of CA1052
+    /// says that the cause of the diagnostic is that the class was not declared sealed. Like
+    /// FxCop, this analyzer does not emit a diagnostic when a non-default constructor is declared,
+    /// even though the title of CA1053 is "Static holder types should not have constructors".
+    /// Like FxCop, this analyzer does emit a diagnostic when the type has a private default
+    /// constructor, even though the documentation of CA1053 says it should only trigger for public
+    /// or protected default constructor.
+    /// </para>
+    /// <para>
+    /// The rationale for all of this is to facilitate a smooth transition from FxCop rules to the
+    /// corresponding Roslyn-based analyzers.
+    /// </para>
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class CA1052DiagnosticAnalyzer : AbstractNamedTypeAnalyzer
+    {
+        public const string DiagnosticId = "CA1052";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(
+            nameof(AnalyzerPowerPackRulesResources.StaticHolderTypesShouldBeStaticOrNotInheritable),
+            AnalyzerPowerPackRulesResources.ResourceManager,
+            typeof(AnalyzerPowerPackRulesResources));
+
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(
+            nameof(AnalyzerPowerPackRulesResources.StaticHolderTypeIsNotStatic),
+            AnalyzerPowerPackRulesResources.ResourceManager,
+            typeof(AnalyzerPowerPackRulesResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            AnalyzerPowerPackDiagnosticCategory.Design,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            helpLinkUri: "http://msdn.microsoft.com/library/ms182168.aspx",
+            customTags: DiagnosticCustomTags.Microsoft);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void AnalyzeSymbol(
+            INamedTypeSymbol symbol,
+            Compilation compilation,
+            Action<Diagnostic> addDiagnostic,
+            AnalyzerOptions options,
+            CancellationToken cancellationToken)
+        {
+            if (symbol.IsStaticHolderType()
+                && !symbol.IsStatic
+                && (symbol.IsPublic() || symbol.IsProtected()))
+            {
```

Comment:
Thanks Charles. Since this is already merged, I will take your feedback in a separate PR. Thanks for looking!